ote/src/cljs/ote/communication.cljs

```diff
@@ -14,32 +17,47 @@
 
 (defn transit-response-format []
   (ajax/transit-response-format {:reader (t/reader :json transit/read-options)
-                                 :raw true}))
+                                 :raw    true}))
+
+(defn- check-progress! []
+  (when (zero? @query-counter)
+    (progress/done)))
+
+(defn- handle-response! []
+  (progress/inc)
+  (swap! query-counter dec)
+  (check-progress!))
 
 (defn- request-url [url]
   (str @base-url url))
 
 (defn get!
   "Make a GET request to the given URL.
   URL parameters can be given with the `:params` key.
-  Callbacks for successfull and failure are provided with `:on-success` and `:on-failure`
+  Callbacks for successful and failure are provided with `:on-success` and `:on-failure`
   keys respectively"
   [url {:keys [params on-success on-failure response-format]}]
+
+  (progress/start)
+  (swap! query-counter inc)
   (GET (request-url url)
-       {:params params
-        :handler on-success
-        :error-handler on-failure
+       {:params          params
+        :handler         (comp handle-response! on-success)
+        :error-handler   (comp handle-response! on-failure)
         :response-format (or response-format (transit-response-format))}))
 
 (defn post!
   "Make a POST request to the given URL.
   URL parameters can be given with the `:body` key.
-  Callbacks for successfull and failure are provided with `:on-success` and `:on-failure`
+  Callbacks for successful and failure are provided with `:on-success` and `:on-failure`
   keys respectively"
   [url body {:keys [on-success on-failure response-format]}]
+
+  (progress/start)
+  (swap! query-counter inc)
   (POST (request-url url)
-        {:params body
-         :handler on-success
-         :error-handler on-failure
-         :format (transit-request-format)
+        {:params          body
+         :handler         (comp handle-response! on-success)
```

Comment:
Minor issue.
While `comp` works, `handle-response!` doesn't take any parameters. This implies that handle-response takes the return value of on-success/on-failure.

I think it would be clearer if handle-response takes the success/failure as parameter and returns a handler which calls the success/failure as the last step.